### PR TITLE
Add painkillers

### PR DESCRIPTION
**Pull request by alganthe**
**When merged this pull request will:**
- Add painkillers

This will need some tests for balance, but it works.
Also missing an inventory icon but i'm terrible at that :P

### CODE REVIEW

File: addons/painkillers/$PBOPREFIX$ [hunk 1]
```diff
@@ -0,0 +1 @@
+x\tac\addons\painkillers
```
Comment (jonpas) [inline addons/painkillers/$PBOPREFIX$]: Missing empty newline.

File: addons/painkillers/ACE_Medical_Actions.hpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class ACE_Medical_Actions {
+    class Basic {
+        class Morphine;
+        class PainKillers: Morphine {
```
Comment (jonpas) [inline addons/painkillers/ACE_Medical_Actions.hpp]: Can we prefix this? Would be safer if/when ACE3 adds it so we can just swap items we use. Also `Painkillers`.

File: addons/painkillers/ACE_Medical_Actions.hpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            displayName = CSTRING(DisplayName);
+            displayNameProgress = CSTRING(painkillerProgress);
+            allowedSelections[] = {"head"};
+            items[] = {"TAC_painkillers"};
```
Comment (jonpas) [inline addons/painkillers/ACE_Medical_Actions.hpp]: Hmm, `tac_painkillers` would be better. But `QUOTE(painkillers)` with component named `tac_medical` would be even better I think.

File: addons/painkillers/CfgVehicles.hpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        };
+    };
+
+    class Man;
+    class CAManBase: Man {
+        class ACE_SelfActions {
+            class Medical {
+                class ACE_Head {
+                    class FieldDressing;
+
```
Comment (jonpas) [inline addons/painkillers/CfgVehicles.hpp]: No empty line.

File: addons/painkillers/CfgVehicles.hpp [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                        icon = "";
+                    };
+                };
+            };
+        };
+
+        class ACE_Actions {
+            class ACE_Head {
+                class FieldDressing;
+
```
Comment (jonpas) [inline addons/painkillers/CfgVehicles.hpp]: No empty line.

File: addons/medical/CfgVehicles.hpp [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
-    class TAC_painkillersItem: Item_Base_F {
+    class GVAR(painkillersItem): Item_Base_F {
         scope = 2;
         scopeCurator = 2;
         displayName = CSTRING(displayName);
         author = "Alganthe";
         vehicleClass = "Items";
         class TransportItems {
-            MACRO_ADDITEM(TAC_painkillers,1);
+            MACRO_ADDITEM(TAC_medical_painkillers,1);
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]: `GVAR(painkillers)`

File: addons/medical/CfgVehicles.hpp [hunk 7]
```diff
@@ -17,12 +17,11 @@ class CfgVehicles {
             class Medical {
                 class ACE_Head {
                     class FieldDressing;
-
-                    class PainKillers: FieldDressing {
+                    class GVAR(painKillers): FieldDressing {
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]: `painkillers`

File: addons/medical/CfgVehicles.hpp [hunk 8]
```diff
@@ -32,12 +31,11 @@ class CfgVehicles {
         class ACE_Actions {
             class ACE_Head {
                 class FieldDressing;
-
-                class PainKillers: FieldDressing {
+                class GVAR(painKillers): FieldDressing {
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]: `painkillers`

**Comment by JoramD0** [comment]
What type of painkiller did you base this on? It looks like (the config at least) it increases blood pressure.

**Comment by alganthe** [comment]
This was based off morphine (ACE's) and no specific "real world" painkiller.

File: addons/medical/ACE_Medical_Actions.hpp [hunk 9]
```diff
@@ -0,0 +1,34 @@
+class ACE_Medical_Actions {
+    class Basic {
+        class Morphine;
+        class GVAR(painkillers): Morphine {
+            displayName = CSTRING(DisplayName);
+            displayNameProgress = CSTRING(painkillerProgress);
```
Comment (jonpas) [inline addons/medical/ACE_Medical_Actions.hpp]:
```suggestion
            displayNameProgress = CSTRING(PainkillerProgress);
```

File: addons/medical/ACE_Medical_Actions.hpp [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+            allowedSelections[] = {"head"};
+            items[] = {QGVAR(painkillers)};
+        };
+    };
+    class Advanced {
+        class Morphine;
+        class GVAR(painkillers): Morphine {
+            displayName = CSTRING(DisplayName);
+            displayNameProgress = CSTRING(painkillerProgress);
```
Comment (jonpas) [inline addons/medical/ACE_Medical_Actions.hpp]:
```suggestion
            displayNameProgress = CSTRING(PainkillerProgress);
```

File: addons/medical/CfgVehicles.hpp [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    };
+
+    class Man;
+    class CAManBase: Man {
+        class ACE_SelfActions {
+            class Medical {
+                class ACE_Head {
+                    class FieldDressing;
+                    class GVAR(painkillers): FieldDressing {
+                        displayName = CSTRING(selfActionDisplayName);
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]:
```suggestion
                        displayName = CSTRING(PainkillerSelfActionDisplayName);
```

File: addons/medical/CfgVehicles.hpp [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+                    };
+                };
+            };
+        };
+
+        class ACE_Actions {
+            class ACE_Head {
+                class FieldDressing;
+                class GVAR(painkillers): FieldDressing {
+                    displayName = CSTRING(actionDisplayName);
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]:
```suggestion
                    displayName = CSTRING(PainkillerActionDisplayName);
```

File: addons/medical/CfgVehicles.hpp [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+    class Man;
+    class CAManBase: Man {
+        class ACE_SelfActions {
+            class Medical {
+                class ACE_Head {
+                    class FieldDressing;
+                    class GVAR(painkillers): FieldDressing {
+                        displayName = CSTRING(selfActionDisplayName);
+                        condition = QUOTE([ARR_4(_player, _target, 'head', QQGVAR(painkillers))] call ace_medical_fnc_canTreatCached);
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]:
```suggestion
                        condition = '[_player, _target, 'head', QGVAR(painkillers)] call ace_medical_fnc_canTreatCached';
```

File: addons/medical/CfgVehicles.hpp [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+                        exceptions[] = {"isNotInside", "isNotSwimming"};
+                        statement = QUOTE([ARR_4(_player, _target, 'head', QQGVAR(painkillers))] call ace_medical_fnc_treatment);
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]:
```suggestion
                        statement = '[_player, _target, 'head', QGVAR(painkillers)] call ace_medical_fnc_treatment';
```

File: addons/medical/CfgVehicles.hpp [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+                };
+            };
+        };
+
+        class ACE_Actions {
+            class ACE_Head {
+                class FieldDressing;
+                class GVAR(painkillers): FieldDressing {
+                    displayName = CSTRING(actionDisplayName);
+                    condition = QUOTE([ARR_4(_player, _target, 'head', QQGVAR(painkillers))] call ace_medical_fnc_canTreatCached);
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]:
```suggestion
                    condition = '[_player, _target, 'head', QGVAR(painkillers)] call ace_medical_fnc_canTreatCached';
```

File: addons/medical/CfgVehicles.hpp [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+                    exceptions[] = {"isNotInside", "isNotSwimming"};
+                    statement = QUOTE([ARR_4(_player, _target, 'head', QQGVAR(painkillers))] call ace_medical_fnc_treatment);
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]:
```suggestion
                    statement = '[_player, _target, 'head', QGVAR(painkillers)] call ace_medical_fnc_treatment';
```

File: addons/medical/CfgWeapons.hpp [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+
+    class GVAR(painkillers): ACE_ItemCore {
+        scope = 2;
+        author = "Alganthe";
+        model = "\A3\Structures_F_EPA\Items\Medical\PainKillers_F.p3d";
+        picture = QPATHTOF(data\painkillers_inventory.paa);
+        displayName = CSTRING(displayName);
+        descriptionShort = "$STR_ACE_Medical_Morphine_Desc_Short";
+        descriptionUse = "$STR_ACE_Medical_Morphine_Desc_Use";
+        class ItemInfo: CBA_MiscItem_ItemInfo {
```
Comment (jonpas) [inline addons/medical/CfgWeapons.hpp]: Empty line above.

File: addons/medical/CfgWeapons.hpp [hunk 18]
```diff
@@ -0,0 +1,17 @@
+class CfgWeapons {
+    class CBA_MiscItem_ItemInfo;
+    class ACE_ItemCore;
+
+    class GVAR(painkillers): ACE_ItemCore {
+        scope = 2;
+        author = "Alganthe";
+        model = "\A3\Structures_F_EPA\Items\Medical\PainKillers_F.p3d";
+        picture = QPATHTOF(data\painkillers_inventory.paa);
+        displayName = CSTRING(displayName);
```
Comment (jonpas) [inline addons/medical/CfgWeapons.hpp]:
```suggestion
        displayName = CSTRING(DisplayName);
```

File: addons/medical/README.md [hunk 19]
```diff
@@ -0,0 +1,11 @@
+# About
+
+Add painkillers, dependent on ACE Medical
```
Comment (jonpas) [inline addons/medical/README.md]:
```suggestion
Adds painkillers to ACE3 Medical.
```
Basic as well, or only Advanced?

File: addons/medical/README.md [hunk 20]
```diff
@@ -0,0 +1,11 @@
...
+
+### Authors
+
+- [Alganthe](http://github.com/alganthe)
+
+### Features
```
Comment (jonpas) [inline addons/medical/README.md]: Remove this section.

File: addons/medical/stringtable.xml [hunk 21]
```diff
@@ -0,0 +1,17 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project name="TAC">
+    <Package name="Medical">
+        <Key ID="STR_TAC_Medical_DisplayName">
+            <English>Painkillers</English>
+        </Key>
+        <Key ID="STR_TAC_Medical_selfActionDisplayName">
```
Comment (jonpas) [inline addons/medical/stringtable.xml]:
```suggestion
        <Key ID="STR_TAC_Medical_PainkillersSelfActionDisplayName">
```

File: addons/medical/stringtable.xml [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+            <English>Take painkillers</English>
+        </Key>
+        <Key ID="STR_TAC_Medical_actionDisplayName">
```
Comment (jonpas) [inline addons/medical/stringtable.xml]:
```suggestion
        <Key ID="STR_TAC_Medical_PainkillersActionDisplayName">
```

File: addons/medical/stringtable.xml [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+            <English>Give painkillers</English>
+        </Key>
+        <Key ID="STR_TAC_Medical_painkillerProgress">
```
Comment (jonpas) [inline addons/medical/stringtable.xml]:
```suggestion
        <Key ID="STR_TAC_Medical_PainkillerProgress">
```

File: addons/medical/CfgWeapons.hpp [hunk 24]
```diff
@@ -0,0 +1,17 @@
+class CfgWeapons {
+    class CBA_MiscItem_ItemInfo;
+    class ACE_ItemCore;
+
+    class GVAR(painkillers): ACE_ItemCore {
+        scope = 2;
+        author = "Alganthe";
+        model = "\A3\Structures_F_EPA\Items\Medical\PainKillers_F.p3d";
+        picture = QPATHTOF(data\painkillers_inventory.paa);
```
Comment (jonpas) [inline addons/medical/CfgWeapons.hpp]: Should be in `ui\painkillers_inventory_ca.paa` (or `_co` if it has no transparency).

File: addons/medical/CfgWeapons.hpp [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+        displayName = CSTRING(displayName);
+        descriptionShort = "$STR_ACE_Medical_Morphine_Desc_Short";
```
Comment (jonpas) [inline addons/medical/CfgWeapons.hpp]:
```suggestion
        descriptionShort = ACECSTRING(Medical,Morphine_Desc_Short);
```

File: addons/medical/CfgWeapons.hpp [hunk 26]
```diff
@@ -0,0 +1,17 @@
...
+        descriptionUse = "$STR_ACE_Medical_Morphine_Desc_Use";
```
Comment (jonpas) [inline addons/medical/CfgWeapons.hpp]:
```suggestion
        descriptionUse = ACECSTRING(Medical,Morphine_Desc_Use);
```

File: addons/medical/CfgVehicles.hpp [hunk 27]
```diff
@@ -0,0 +1,44 @@
...
+                    icon = QPATHTOF(data\painkillers_icon.paa);
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]: Should be in `ui\painkillers_icon_ca.paa` (or `_co` if it has no transparency).

File: addons/medical/CfgVehicles.hpp [hunk 28]
```diff
@@ -0,0 +1,45 @@
...
+
+    class Man;
+    class CAManBase: Man {
+        class ACE_SelfActions {
+            class Medical {
+                class ACE_Head {
+                    class FieldDressing;
+                    class GVAR(painkillers): FieldDressing {
+                        displayName = CSTRING(PainkillersSelfActionDisplayName);
+                        condition = "[_player, _target, 'head', 'tac_medical_painkillers'] call ace_medical_fnc_canTreatCached";
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]:
```suggestion
                        condition = '[_player, _target, "head", QGVAR(painkillers)] call ACEFUNC(medical,canTreatCached)';
```

File: addons/medical/CfgVehicles.hpp [hunk 29]
```diff
@@ -0,0 +1,45 @@
...
+                        exceptions[] = {"isNotInside", "isNotSwimming"};
+                        statement = "[_player, _target, 'head', 'tac_medical_painkillers'] call ace_medical_fnc_treatment";
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]:
```suggestion
                        statement = '[_player, _target, "head", QGVAR(painkillers)] call ACEFUNC(medical,treatment)';
```

File: addons/medical/CfgVehicles.hpp [hunk 30]
```diff
@@ -0,0 +1,45 @@
...
+                };
+            };
+        };
+
+        class ACE_Actions {
+            class ACE_Head {
+                class FieldDressing;
+                class GVAR(painkillers): FieldDressing {
+                    displayName = CSTRING(PainkillersActionDisplayName);
+                    condition = "[_player, _target, 'head', 'tac_medical_painkillers'] call ace_medical_fnc_canTreatCached";
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]:
```suggestion
                    condition = '[_player, _target, "head", QGVAR(painkillers)] call ACEFUNC(medical,canTreatCached)';
```

File: addons/medical/CfgVehicles.hpp [hunk 31]
```diff
@@ -0,0 +1,45 @@
...
+                    exceptions[] = {"isNotInside", "isNotSwimming"};
+                    statement = "[_player, _target, 'head', 'tac_medical_painkillers'] call ace_medical_fnc_treatment";
```
Comment (jonpas) [inline addons/medical/CfgVehicles.hpp]:
```suggestion
                    statement = '[_player, _target, "head", QGVAR(painkillers)] call ACEFUNC(medical,treatment)';
```

File: addons/medical/CfgWeapons.hpp [hunk 32]
```diff
@@ -0,0 +1,18 @@
+class CfgWeapons {
+    class CBA_MiscItem_ItemInfo;
+    class ACE_ItemCore;
+
+    class GVAR(painkillers): ACE_ItemCore {
+        scope = 2;
+        author = "Alganthe";
+        model = "\A3\Structures_F_EPA\Items\Medical\PainKillers_F.p3d";
+        picture = QPATHTOF(UI\painkillers_inventory_ca.paa);
```
Comment (jonpas) [inline addons/medical/CfgWeapons.hpp]:
```suggestion
        picture = QPATHTOF(ui\painkillers_inventory_ca.paa);
```

**Pull request merged**